chromium/preloadlist/preloadlist.go

```diff
@@ -62,12 +80,27 @@ func (p PreloadList) Index() (idx IndexedEntries) {
 	}
 }
 
-// Get acts similar to map access: it returns an entry from the index preload
-// list (if it is present), along with a boolean indicating if the entry is
-// present.
-func (idx IndexedEntries) Get(domain string) (Entry, bool) {
-	entry, ok := idx.index[strings.ToLower(domain)]
-	return entry, ok
+// Get returns an entry from the index preload list along with a status
+// indicating how the entry is found. If the domain itself is on the preload
+// list, its entry is returned. If one of its ancestor domain is on the
+// list, and "include_subdomains" is set on that ancestor domain, the ancestor
+// entry is return. Failing all that, a zero-value entry is returned.
+func (idx IndexedEntries) Get(domain string) (Entry, HstsPreloadStatus) {
+	// Check if the domain itself is on the list.
+	domain = strings.ToLower(domain)
+	entry, ok := idx.index[domain]
+	if ok {
+		return entry, ExactEntryFound
+	}
+	// Walk up the chain until we find an ancestor domain which includes subdomains.
+	for i := strings.Index(domain, "."); i != -1 && i != len(domain)-1; domain = domain[i+1:] {
```

Comment:
Nit: the intent is clear, but the code is not easy to understand at a glance.

I would write a function `parentDomain()` and call:

    while ancestor, ok := parentDomain(ancestor); ok {
      // ...
    }